src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -3711,60 +3723,133 @@ subroutine bkgcov_a_en_new_factorization(ig,a_en)
   type(gsi_bundle),intent(inout) :: a_en(n_ens)
 
 ! Local Variables
-  integer(i_kind) ii,k,iadvance,iback,is,ie,ipnt,istatus
+  integer(i_kind) ii,i,j,k,iadvance,iback,is,ie,ipnt,istatus,kk
   real(r_kind) hwork(grd_loc%inner_vars,grd_loc%nlat,grd_loc%nlon,grd_loc%kbegin_loc:grd_loc%kend_alloc)
+  real(r_kind) work_tmp(grd_loc%lon2,grd_loc%lat2)
```

Comment:
work_tmp could be defined as allocatable arrays since it is only used when l_mbgf_loc =.true.